chia/data_layer/s3_plugin_service.py

```diff
@@ -233,7 +254,10 @@ async def download(self, request: web.Request) -> web.Response:
 
             bucket_str = parse_result.netloc
             my_bucket = self.boto_resource.Bucket(bucket_str)
-            target_filename = self.server_files_path.joinpath(filename)
+            if group_files_by_store:
+                target_filename = self.server_files_path.joinpath(f"{filename_store_id}").joinpath(filename[65:])
+            else:
+                target_filename = self.server_files_path.joinpath(filename)
```

Comment:
This file seems a bit more duplicated than the datalayer itself in terms of defining the file name forms.  Could a function be made for this?